Sources/General/Commands/Setup.swift

```diff
@@ -0,0 +1,322 @@
+        spec.target = ask("Target (optional)")
+        spec.testTarget = ask("Test target (optional)")
+        spec.company = ask("Company (optional)", default: spec.company)
+        guard let data = try? specFactory.makeData(spec: spec) else {
+            return false
+        }
+        do {
+            try data.write(to: destination)
+            return true
+        }
+        catch {
+            return false
+        }
+    }
+
+    private func askBool(question: String) -> Bool {
+        guard let result = ask(question) else {
+            return askBool(question: question)
+        }
+        switch result.lowercased() {
+        case "yes", "y":
+            return true
+        case "no", "n":
+            return false
+        default:
+            return askBool(question: question)
+        }
+    }
+
+    private func ask(_ question: String, default: String? = nil) -> String? {
+        if let value = `default` {
+            print("\(question) \u{001B}[0;32m(\(value))\u{001B}[0;0m:", terminator: " ")
+        }
+        else {
+            print("\(question):", terminator: " ")
+        }
+        guard let value = readLine(),
+            value.isEmpty == false else {
+            return `default`
+        }
+        return value
+    }
+
+    private func findProject() -> String? {
+        let info = fileHelper.contentsOfDirectory(at: Constants.relativeCurrentPath).first { info in
+            info.url.pathExtension == Constants.xcodeProjectPathExtension
+        }
+        return info?.url.lastPathComponent
+    }
+
+    private func remove(_ url: URL) throws {
+        do {
+            try fileHelper.fileManager.removeItem(at: url)
+        }
+        catch {
+            throw Error.remove(url)
+        }
+    }
+
+    private func displayResult(_ templates: [FileInfo], isSpecModified: Bool) {
+        if templates.isEmpty {
+            print("\u{001B}[0;33mNo templates modified 🤷‍♂️")
+        }
+        else {
+            print("✨ Updated templates:")
+            templates.forEach { file in
+                print("\u{001B}[0;32m" + file.url.lastPathComponent)
+            }
+
+        }
+        if isSpecModified {
+            print("\n\u{001B}[0;32mGeneral spec modified")
+        }
+        print("\u{001B}[0;0m")
+    }
+}
+
+extension Setup.Error: CustomStringConvertible {
+
+    var description: String {
+        switch self {
+        case .githubName(let github):
+            return "Could not retrieve templates url from provided github (\(github)"
+        case .url(let url):
+            return "Invalid url provided \(url)"
+        case .download(let url):
+            return "Cold not download teplates from url \(url)"
+        case .write(let destination):
+            return "Could not write templates to their destination \(destination)"
+        case .remove(let url):
+            return "Could not remove temp directory at \(url)"
+        }
+    }
+}
```

Comment:
Can we place it to where `Setup.Error` is defined?

Reply:
https://github.com/rosberry/general/pull/1#issuecomment-689453692